### ISSUE TITLE
Add to Favorites in player (streaming service)

### ISSUE THREAD

**Issue by larod241** [LMS-Community/slimserver#744]
Hello, 

It is possible to create an API to add favourites to a service, detect if item is favourited, remove favourite, etc.

The goal is to be able to manage the favorites of the streaming services (Qobuz and Spotify for my part) with a heart that opens directly the menu for managing the favorites of the associated service. (see gif).

I asked @CDrummond (skin material developer) but it seems that the problem is on the LMS side and that an API is needed. 
[https://github.com/CDrummond/lms-material/issues/551](url)

I will give him that outcome so that he can complete it. 

Currently, this is an important feature that would make the use much easier/enjoyable! 

Thank you all again.

![fav](https://user-images.githubusercontent.com/20423481/148119293-2c9dbad3-4b52-4c31-b1ce-850ac2c43c7d.gif)

**Comment by mherger** [comment]
This would need some work on the music service integrations, too. In particular Qobuz is rather short on those features... and I don't have access to their API documentation any more.

**Comment by Capstan62** [comment]
Hi Gents,

I agree with [larod241] about the importance of this enhancement. In the past I was not sure if it was more a Material Skin limitation or a gap in the Qobuz plugin integration in LMS. I am using LMS in Daphile. I wrote a message to Qobuz plugin authors a while ago (don't remember if Mr. Herger was the recipient, sorry) and - as far as I remember - they replied saying they received a lot of requests for this but they could not commit on this on a short time period.
So I am living with it managing favorites from my Qobuz phone app.
However today I was playing with iPeng (and app for controlling LMS from iPad and Mac) and to my surprise you can add favorites from within the application (you can add only, not remove from what I see). This means the Qobuz plugin in LMS should be  already capable, possibly shifting the gap to Material Skin interface only.
In other words, if iPeng can do it, I see no reason Material could not.

I take the occasion for sending Mr Herger a big thank you for the nice work done with the Qobuz plugin.

My 2 cents,
Ale, Milano.

**Comment by Capstan62** [comment]
EDIT: [just]([url](url)) one more note. The save favorite feature is indeed present in iPeng, although in some instance it is not. I don't understand the rationale.
Say, I search for an album of Francis Cabrel, found, long click on the album picture then in the pop-up menu "Add "Album name" to favorites" appears.
In an other intense I search for "432HZ celestial" (don't blame me, it's a relaxing music I use to sleep LOL), found it, but the pop-up menu does not contain same option as above. Weird...
<img width="1230" alt="Schermata 2022-08-23 alle 08 33 21" src="https://user-images.githubusercontent.com/79332801/186088934-2fb3c89d-8bd1-4f87-b6b3-ef94f12b0f69.png">

<img width="1235" alt="Schermata 2022-08-23 alle 08 30 04" src="https://user-images.githubusercontent.com/79332801/186089001-fbc54d22-43de-418b-b4a2-87818397d94b.png">

**Comment by CDrummond** [comment]
@Capstan62 The menu you show in your screenshot corresponds to those from the "(...) More" menus in Material - so that 'Add XXX to favorites' should already be there (the menu is created by LMS). But, AFAIK, it would be adding to LMS's favourites, not to the streaming service's favourites. This issue (AFAICT) is about adding it to a streaming service's list of favourites, not LMS.

**Comment by Capstan62** [comment]
Hi Craig,

first of all a big thank you for the amazing work you have done on Material Skin for LMS. I absolutely love it!

Now, back at our conversation.
You are (of course) right. A More option is available in Material UI and it does add a Favorite item but in LMS only (not in the Streaming Service) and at single track level, not Album (which is what I am looking for).

I have done again the exercise in iPeng and I confirm it does add an entire album in the Qobuz Favorite repository. See and believe.

First screenshot is the process I followed in iPeng. 
Search artist, select album (long press), select the add to favorite option in the pop-up menu.

Then I have checked in the Qobuz mobile app and within few seconds the item appeared in my Favorite list.

Thank you for your attention.
Alessandro.

<img width="878" alt="Schermata 2022-08-23 alle 10 46 36" src="https://user-images.githubusercontent.com/79332801/186116048-bff40397-f7bc-4779-96ae-feca5e47b87a.png">

![IMG_8409](https://user-images.githubusercontent.com/79332801/186116447-13c96668-2fd5-491d-a4eb-09037aa4d507.PNG)
